product_logistics_uom/models/product_product.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2023 ACSONE SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import api, fields, models
+from odoo.tools import float_is_zero
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    # remove rounding from volume and weight
+    # this is needed to avoid rounding errors when converting between units
+    # and is safe since we display the volume and weight in the product's
+    # volume and weight UOM. In the same time, we need to keep the volume
+    # we ensure that no information is lost by storing the volume and weight
+    # without rounding.
+    volume = fields.Float(digits=False)
```

Comment:
sorry for make noise, you can keep that code or set digits=None. because we are overrided this attribute https://github.com/odoo/odoo/blob/fa90f11227591ed185a091092aa913080b73e222/addons/product/models/product_template.py#L105